gwcs/wcstools.py

```diff
@@ -271,10 +273,83 @@ def wcs_from_points(xy, world_coordinates, fiducial,
         poly_x = fitter(poly, x, y, projection_x)
         poly_y = fitter(poly, x, y, projection_y)
     transform = models.Mapping((0, 1, 0, 1)) | poly_x & poly_y | projection.inverse | skyrot.inverse
-    
+
     skyframe = CelestialFrame(reference_frame=fiducial.frame)
     detector = Frame2D(name="detector")
     pipeline = [(detector, transform),
                 (skyframe, None)
                 ]
     return WCS(pipeline)
+
+
+def fit_wcs_from_points(xy, world_coordinates, proj_point=None, projection=None,
+                        degree=4, distortion_type='polynomial'):
+    """
+    Given two matching sets of coordinates on detector and sky, fit a WCS.
+
+    Notes
+    -----
+    This function implements the following algorithm:
+    ``world_coordinates`` are transformed to a projection plane using the specified projection.
+    A polynomial fits ``xy`` and the projected coordinates.
+    The fitted polynomials and the projection transforms are combined into a tranform
+    from detector to sky.
+    The input coordinate frame is set to ``detector``.
+    The output coordinate frame is initialized based on the frame in the fiducial.
+
+
+    Parameters
+    ----------
+    xy : tuple of 2 ndarrays
+        Points in the input cooridnate frame - x, y inputs.
+    world_coordinates : `~astropy.coordinates.SkyCoord` or tuple of 2 ndarrays
+        Points in the output coordinate frame.
+        The order matches the order of ``xy``.
+        If a tuple, it is assumed the numbers represent location in the same frame as `proj_point`.
+    proj_point : `~astropy.coordinates.SkyCoord` or str
+        A fiducial point in the output coordinate frame.
+    projection : `~astropy.modeling.projections.Projection`
```

Comment:
Again, the actual default value (`None`) used in the function declaration is different from the allowed types/values listed here. Please add `None` to the list and explain what is it doing. Looking at the code, I think it would be better (more explicit) not to have `None` at all and instead simply set `projection=projections.Sky2Pix_TAN()` as it was done before.

Reply:
Moreover, looking at the code, I can see `projection` can be of string type. Please, either remove that code or adjust docs to include `str` type and document what values for this string type are allowed.